### PR TITLE
[tree_sitter] Add sum type to tree node

### PR DESCRIPTION
**Pull request by Lycs-D**

### CODE REVIEW

File: tree_sitter_v/grammar.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ const list_separator = choice(semi, ',');
 module.exports = grammar({
 	name: 'v',
 
-	extras: ($) => [/\s/, $.line_comment, $.block_comment],
+	extras: ($) => [/[ \t]+|(\r?\n)+/, $.line_comment, $.block_comment],
```
Comment (ttytm) [inline tree_sitter_v/grammar.js:86]: If possible, please explain why this change

File: tree_sitter_v/package.json [hunk 2]
```diff
@@ -16,8 +16,8 @@
 		"install": "node-gyp-build",
 		"prebuildify": "prebuildify --napi --strip",
 		"lint": "eslint \"**/*.js\"",
-		"format": "prettier --write \"**/*.js\"",
-		"format:check": "prettier --check \"**/*.js\""
+		"format": "prettier --write \"grammar.js\"",
+		"format:check": "prettier --check \"grammar.js\""
```
Comment (ttytm) [inline tree_sitter_v/package.json:20]: We have multiple js files in that directory. Prettier can take care of all of them.

**Review by ttytm — COMMENTED** [review]
Thanks for the PR!

File: tree_sitter_v/grammar.js (hunk 1 above)
Comment (JalonSolov) [reply tree_sitter_v/grammar.js:86]: Why won't `\s+` work?  The added `+` seems to be the only real change, except that yours covers fewer characters...
```
The \s metacharacter matches a whitespace character.

Whitespace characters can be:

     * A space character
     * A tab character
     * A carriage return character
     * A new line character
     * A vertical tab character
     * A form feed character
```

**Review by ttytm — APPROVED** [review]
